python_ta/upload.py

```diff
@@ -0,0 +1,87 @@
+import uuid
+import hashlib
+import requests
+import json
+from typing import Dict
+
+
+def errors_to_dict(errors):
+    """Convert PyTA errors from MessageSet format to json."""
+    error_info = ['msg_id', 'msg', 'symbol', 'module', 'category', 'line']
+    error_types = ['code', 'style']
+    err_as_dict = {}
+    for msg_set in errors:  # This iterates over the (filename, code, style) MessageSets
+        for error_type in error_types:  # This iterates over the code and style attributes
+            current_type = getattr(msg_set, error_type)  # Gets either the code or style dictionary
+            for key in current_type.keys():  # Iterates over the error id's of caught errors
+                err_as_dict[key] = []
+                info_set = current_type.get(key)
+                for msg in info_set.messages:  # Iterates over the messages for each error of the given code
+                    info = {k: getattr(msg, k) for k in error_info}
+                    err_as_dict[key].append(info)
+    return err_as_dict
+
+
+def upload_to_server(errors, config: Dict[str, str], default: Dict[str, str], url: str, version: str, paths=None):
```

Comment:
There are more type annotations to add here. You can break the header into two lines.